src/role_nextroomer.js

```diff
@@ -308,11 +308,15 @@ roles.nextroomer.settle = function(creep) {
     methods.push(Creep.constructTask);
   }
 
+  // Fill energy after building is done
+  if (creep.room.controller.level >= 4) {
+    methods.push(Creep.transferEnergy);
+  }
+
   if (creep.room.controller.level < 8) {
+
```

Comment:
As far as i understand the nextroomer role.
Build up a new room to a certain level or revive/rebuild a fallen room

with the other checks they will upgrade less after RCL4 i reached. maybe we can remove the check.